internal/providermeta/meta.go

```diff
@@ -68,11 +72,38 @@ func LoadApplicationURL(ctx context.Context, meta any, fragments ...string) stri
 	return u.String()
 }
 
-func (s *Meta) Validate() (errs error) {
-	if s.AuthToken == "" {
-		errs = multierr.Append(errs, errors.New("auth token not set"))
+// LoadSessionToken will use the provider username and password
+// so that it can be used as the token through the interaction.
+func (m *Meta) LoadSessionToken(ctx context.Context) (string, error) {
+	if m.AuthToken != "" {
+		return m.AuthToken, nil
 	}
-	if s.APIURL == "" {
+
+	client, err := signalfx.NewClient("", signalfx.APIUrl(m.APIURL))
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := client.CreateSessionToken(ctx, &sessiontoken.CreateTokenRequest{
+		Email:          m.Email,
+		Password:       m.Password,
+		OrganizationId: m.OrganizationID,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	// TODO: determine if any additional fields would be useful for debugging.
+	tflog.Info(ctx, "Created new session token")
+
+	return resp.AccessToken, nil
+}
+
+func (m *Meta) Validate() (errs error) {
+	if m.AuthToken == "" && (m.Email == "" && m.Password == "") {
```

Comment:
If either email OR password are missing, credentials are incomplete so validation should fail.
`if m.AuthToken == "" && (m.Email == "" || m.Password == "")`

Reply:
Included some test cases to cover this case as well.